furnace.c

```diff
@@ -227,14 +248,103 @@ set_max_pwm_safe(furnace_context_t *ctx, int new_max_pwm)
 static void
 command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const char*, const size_t))
 {
-  unsigned arg;
+  unsigned arg, arg2;
   char     str_arg[BUF_SIZE];
 
   if (buffer[0] == '\n') return;
 
   if (memcmp(buffer, "reboot", 6) == 0) {
     reset_usb_boot(0,0);
-  } else if (sscanf(buffer, "max_pwm %u", &arg) == 1) {
+  } 
+#if CONFIG_SPIN_COATER
+#if CONFIG_SPIN_COATER_PWM
+  else if (sscanf((const char*)buffer, "spin_pwm %u", &arg) == 1) {
+      const bool res = set_spin_coater_pwm_safe(ctx, arg);
+      if (!res) {
+        DEBUG_printf("Chaning PWM duty failed\n");
+        return;
+      }
+      if (arg == PWM_IDLE_DUTY) {
+        ctx->spin_coater.spin_state = SPIN_IDLE;
+        cancel_alarm(ctx->spin_coater.timer_id);
+        const char* msg = "spin_stopped\n";
+        feedback(msg, strlen(msg));
+      } else {
+        ctx->spin_coater.spin_state = SPIN_STARTED_WITH_FORCE_VALUE;
+      }
+
+      char msg[BUF_SIZE];
+      const size_t msg_len =
+        snprintf(msg, sizeof(msg), "spin_pwm %u\n", ctx->spin_coater.pwm_duty);
+      feedback(msg, msg_len);
```

Comment:
Hmm.. I wonder if it's not too much output for the user. I don't think you are usually interested in those log messages. Is this value any useful if not debugging?

Reply:
I used that for my GUI application to confirm the value was set. but since you don't use GUI, I can change that to log with debug level

Reply:
For GUI it kinda make sense, but for TUI it will just make the output less readable. If this information is of any use to the user, we could print it, but in the standard status line that we send each second with temperature and furnace PWM values.

Reply:
Ok, I removed that completely